cmd/pj-rehearse/main.go

```diff
@@ -143,29 +135,46 @@ func main() {
 		}
 	}
 
-	prowConfig, prowPRConfig, err := getProwConfigs(o.candidatePath, jobSpec.Refs.BaseSHA)
+	prConfig := config.GetAllConfigs(o.releaseRepoPath, logger)
+	masterConfig, err := config.GetAllConfigsFromSHA(o.releaseRepoPath, jobSpec.Refs.BaseSHA, logger)
 	if err != nil {
-		logger.WithError(err).Error("could not load prow configs")
+		logger.WithError(err).Error("could not load configuration from base revision of release repo")
 		gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
-	ciopConfig, ciopPRConfig, err := getCiopConfigs(o.candidatePath, jobSpec.Refs.BaseSHA)
-	if err != nil {
-		logger.WithError(err).Error("could not load ci-operator configs")
+	// We always need both Prow config versions, otherwise we cannot compare them
+	if masterConfig.Prow == nil || prConfig.Prow == nil {
```

Comment:
can we do that inside `GetAllConfigs`?

Reply:
Not really. The decision to `os.Exit` should be certainly kept at the `main()` level, library methods should not `Exit`. Returning a single `error` value with the _at least one thing failed_ semantic was not a good option because we can handle some failures differently (see https://github.com/openshift/ci-operator-prowgen/pull/99#issuecomment-469331170). Hence this: the contract is quite simple (_load all configs, those that fail will be `nil`_) and the callsite knows better what members it needs for what.